reference.module

```diff
@@ -258,17 +258,27 @@ function reference_rendered_formatter_view($entity_type, $entity, $field, $insta
   return $elements;
 }
 
- /**
-  * Implements hook_field_widget_info() to define an autocomplete form widget.
-  */
- function reference_field_widget_info() {
-   return array(
-     'reference_autocomplete' => array(
-       'label' => t('Autocomplete'),
-       'field types' => array('reference'),
-     ),
-   );
- }
+/**
+ * Implements hook_field_widget_info() to define an autocomplete form widget.
+ */
+function reference_field_widget_info() {
+  return array(
+    'reference_autocomplete' => array(
+      'label' => t('Autocomplete'),
+      'field types' => array('reference'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_field_widget_info_alter() to enable select and buttons options.
+ */
+function reference_field_widget_info_alter(&$info) {
+  if (module_exists('options')) {
+    $info['options_select']['field types'][] = 'reference';
+    $info['options_buttons']['field types'][] = 'reference';
+  }
+}
```

Comment:
Can you update this pull request to match the indentation of the original module so that we can compare them more easily?

Reply:
Sure. I guess we need a separate pull request for the styling fix.

Reply:
Done (I think).